src/jcarbon/src/main/java/jcarbon/emissions/EmissionsConverters.java

```diff
@@ -0,0 +1,61 @@
+
+package jcarbon.emissions;
+
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors; 
+import java.util.stream.Stream;
+import java.lang.Integer; 
+
+import jcarbon.cpu.rapl.RaplReading;
+import jcarbon.cpu.rapl.RaplInterval;
+import jcarbon.cpu.eflect.TaskEnergy;
+
+import jcarbon.data.Interval;
+import jcarbon.emissions.EmissionsInterval;
+import jcarbon.emissions.EmissionsConverter;
+
+/** A class that creates a carbon intensity map from locale. */
+public final class EmissionsConverters {
+    ///private static final String DEFAULT_INTENSITY_FILE = "jRAPL/src/jcarbon/src/resources/emissions/WorldIntensity.csv";
+    private static final String DEFAULT_INTENSITY_FILE = System.getProperty("user.dir") + "/jRAPL/src/jcarbon/src/resources/emissions/WorldIntensity.csv";
+    private static final double GLOBAL_INTENSITY = 475.0;
+
+    public static final EmissionsConverter GLOBAL_CONVERTER = new EmissionsConverter(GLOBAL_INTENSITY);
+    private static final Map<String, Double> CARBON_INTENSITY_MAP = getCarbonIntensity();
+
+    public static EmissionsConverter forLocale(String locale){
+        if(CARBON_INTENSITY_MAP.containsKey(locale)){
+            return new EmissionsConverter(CARBON_INTENSITY_MAP.get(locale).doubleValue());
+        }
+        else{
+            return GLOBAL_CONVERTER;
+        }
+        
+    }
+
+    private static Map<String, Double> getCarbonIntensity(){
+        //parses the csv, maps into a Map;
+        Path filepath = Path.of(System.getProperty("jcarbon.emissions.intensity", DEFAULT_INTENSITY_FILE));
+        if(!Files.exists(filepath)){
+            return Map.of();
+        }
+        try {
+            return Files.readAllLines(Path.of(System.getProperty("jcarbon.mix.emissions", DEFAULT_INTENSITY_FILE)))
+                    .stream()
+                    .skip(1)
+                    .collect(Collectors.toMap(s -> s.split(",")[0], s -> Double.parseDouble(s.split(",")[2])));
+        } catch (IOException e) {
```

Comment:
You shouldn't handle exceptions this way. Although you handled `FileNotFound`, there are still other i/o errors that can be swallowed here.

In addition, you definitely don't want to return null. Imagine that this happens and `CARBON_INTENSITY_MAP` is now set to null. All calls to `forLocale` will fail when `containsKey` is called because the object is not a map, so it doesn't have that method, which will throw a `NullPointerException`.

The correct thing to do here is to re-throw the exception. If the file exists and it can't be read, then there is something fundamentally wrong with the runtime and that is not our responsibility. We don't want someone to continue running their program after trying to use our system to load something bad.

Since this code is not guaranteed to survive long term, can you just replace the two lines you have in the catch block and throw a [`RuntimeException`](https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/lang/RuntimeException.html)? Runtime exceptions are unchecked, so you won't have the compiler bothering you. I recommend using an [`IllegalStateException`](https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/lang/IllegalStateException.html). I think we want something like this:

```
throw new IllegalStateException(String.format("Unable to read %s", filepath), e);
```

Reply:
When you say io errors that can be swallowed, does that mean we shouldn't catch an IOException? I am still a bit confused about what you mean by swallowing. I know we wouldn't want to group the cpu's fault on reading files versus the user's mistakes, so are files not existing and invalid files the only two user errors we want to look out for?

Reply:
[Error swallowing](https://en.wikipedia.org/wiki/Error_hiding) is when you catch an error but then do nothing with it. Although you have logged it to the console, there is a future failure associated with `forLocale` that may not appear until long after this code has run that you never handled. Because we already confirmed the file exists, we know (implicitly) that any I/O error is probably caused by something else. As we cannot confirm what this something else is without digging into the exception (something we do not want to do), we instead re-throw and let the user figure it out.